dash/include/dash/halo/Halo.h

```diff
@@ -726,33 +949,36 @@ class HaloBlock {
   using Region_t        = Region<ElementT, PatternT, NumDimensions>;
   using RegionCoords_t  = RegionCoords<NumDimensions>;
   using region_extent_t = typename RegionSpec_t::region_extent_t;
-  using HaloExtsMax_t =
-    std::array<std::pair<region_extent_t, region_extent_t>, NumDimensions>;
 
 public:
   using Element_t = ElementT;
   using Pattern_t = PatternT;
   using GlobMem_t =
     GlobStaticMem<ElementT, dash::allocator::SymmetricAllocator<ElementT>>;
-  using CycleSpec_t     = CycleSpec<NumDimensions>;
+  using GlobBoundSpec_t = GlobalBoundarySpec<NumDimensions>;
   using pattern_size_t  = typename PatternT::size_type;
   using ViewSpec_t      = typename PatternT::viewspec_type;
   using HaloSpec_t      = HaloSpec<NumDimensions>;
   using RegionVector_t  = std::vector<Region_t>;
   using region_index_t  = typename RegionSpec_t::region_index_t;
   using ElementCoords_t = std::array<pattern_index_t, NumDimensions>;
 
+  using HaloExtsMaxPair_t = std::pair<region_extent_t, region_extent_t>;
+  using HaloExtsMax_t     = std::array<HaloExtsMaxPair_t, NumDimensions>;
+
 public:
+  /**
+   * Constructor
+   */
   HaloBlock(GlobMem_t& globmem, const PatternT& pattern, const ViewSpec_t& view,
-            const HaloSpec_t&  halo_reg_spec,
-            const CycleSpec_t& cycle_spec = CycleSpec_t{})
+            const HaloSpec_t&      halo_reg_spec,
```

Comment:
It's entirely not clear to me what happens in this function. Some breakdown into smaller functions or comments would be nice :)